src/display/editor/tools.js

```diff
@@ -59,6 +59,127 @@ class IdManager {
   }
 }
 
+/**
+ * Class to manage the images used by the editors.
+ * The main idea is to try to minimize the memory used by the images.
+ * The images are cached and reused when possible
+ * We use a refCounter to know when an image is not used anymore but we need to
+ * be able to restore an image after a remove+undo, so we keep a file reference
+ * or an url one.
+ */
+class ImageManager {
+  #baseId = window.crypto.randomUUID();
```

Comment:
There's a couple of issues with using this:
- The `crypto.randomUUID()` function is not available everywhere just yet *and* it's also limited to only secure contexts; please see https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID
 - The `crypto.getRandomValues()` function should be genererally available, however in Node.js environments `crypto` itself may not be; please see https://nodejs.org/dist/latest-v20.x/docs/api/crypto.html#determining-if-crypto-support-is-unavailable

I wrote a helper function to safely obtain a uuid, which you're most welcome to use here; please see https://github.com/mozilla/pdf.js/pull/16610/files#diff-820af4a764722bb36f3daf6f9cb815fedfece273073b21d858bb8aa5210e411cR1016-R1036